src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -1534,6 +1540,12 @@ private static FileChange GetFileChangeDetails(Range changeRange, string insertS
             ILogger Logger,
             CancellationToken cancellationToken)
         {
+            // If filesToAnalzye is empty, nothing to do so return early.
+            if (filesToAnalyze.Length == 0)
```

Comment:
This if check seems to be in both places. Here and [L1505](https://github.com/PowerShell/PowerShellEditorServices/pull/789/files#diff-54e1b8c127a6e69bb92d6ebdd738e0e5R1505). One can probably be deleted, right?

Reply:
It is perhaps a bit paranoid but I wanted to avoid an unnecessary `Task.Delay` if there were no files to analyze.  Right now, only the method above (which also checks for 0 files) is the only caller but in the future - who knows.  I can take it out if you prefer.  It's probably a bit too paranoid.  :-)

Reply:
well I think you can keep this one and safely deleted the other. Won't add too much extra work because it never gets to the `Task.Delay` below, right?

I guess it's taking time to create the Task which is a bit of a pain...

Reply:
If nothing else calls this function, I think you should keep the other and get rid of this one actually

Reply:
Agreed on which one is more important. I'll remove this one.